exporter/awsxrayexporter/internal/translator/cause.go

```diff
@@ -486,6 +489,59 @@ func fillDotnetStacktrace(stacktrace string, exceptions []awsxray.Exception) []a
 	return exceptions
 }
 
+func fillGoStacktrace(stacktrace string, exceptions []awsxray.Exception) []awsxray.Exception {
+	var line string
+	var label string
+	var path string
+	var lineNumber int
+
+	r := textproto.NewReader(bufio.NewReader(strings.NewReader(stacktrace)))
+
+	// Skip first line containing top level exception / message
+	_, _ = r.ReadLine()
+	exception := &exceptions[0]
+	line, err := r.ReadLine()
+	if err != nil {
+		return exceptions
+	}
+
+	exception.Stack = make([]awsxray.StackFrame, 0)
+	for {
+		re := regexp.MustCompile(`^goroutine.*\brunning\b.*:$`)
+		match := re.Match([]byte(line))
+		if match {
+			line, _ = r.ReadLine()
+		}
+
+		label = line
+		line, _ = r.ReadLine()
+
+		if strings.Contains(line, "\t") {
+			a := regexp.MustCompile(":")
+			s := a.Split(line, 2)
+			if len(s) >= 2 {
+				path = strings.Trim(s[0], "\t")
+				lineNumber, _ = strconv.Atoi(strings.Split(s[1], " +0x")[0])
+			}
+		}
```

Comment:
```suggestion
		plnre := regexp.MustCompile(`([^:\s]+)\:(\d+)`)
		matches := plnre.FindStringSubmatch(line)
		if len(matches) == 3 {
			path = matches[1]
			lineNumber, _ = strconv.Atoi(matches[2])
		}
```

Again, the regex should be initialized once, this is just for demonstration purposes.  This should be less brittle than chained splits and easier to adapt if our understanding of the format evolves.